### ISSUE TITLE
`pdm init` fails with `[VirtualenvCreateError]: Can't resolve python interpreter`

### ISSUE THREAD

**Issue by suhail-singh** [macro128/pdm-conda#37]
After installing `pdm-conda` via

``` shell
pdm self add pdm-conda
```

`pdm init` fails.  When the below commands are run:

``` shell
cd /tmp
mkdir -p foo && cd foo
pdm init
```

The following failure is observed:

```
Creating a pyproject.toml for PDM...
STATUS: Downloading cpython@3.12.3
Successfully installed cpython@3.12.3
Version: 3.12.3
Executable: /home/${USER}/.rye/py/cpython@3.12.3/bin/python3
ERROR: Failed to install Python /home/${USER}/.rye/py/cpython@3.12.3/bin/python3: Could not find a version matching version='3.12.3/bin/python3', implementation=/home/${USER}/.rye/py/cpython
[VirtualenvCreateError]: Can't resolve python interpreter /home/${USER}/.rye/py/cpython@3.12.3/bin/python3
```

The very same minimal example succeeds as soon as `pdm-conda` is uninstalled
via:

``` shell
pdm self remove pdm-conda
```

**Comment by macro128** [comment]
Hi! Sorry for the delay, if you want to use conda/mamba with the plugin then you should try using `pdm init --conda-runner conda/mamba`.
However it is failing when you use the command without flags, I'll fix that problem and include it in the next version.

**Comment by macro128** [comment]
Hi! Apart from using `--conda-runner` flag to use conda related environment, using `pdm init` was fixed on v0.18.1
